### ISSUE TITLE
Support integration with Whisper to generate external captions for youtube videos

### ISSUE THREAD

**Issue by jbfletch** [FrostCo/AdvancedProfanityFilter#500]
#### :loudspeaker: Audio Muting will not work as YouTube removes profanity from transcripts
YouTube automatically blocks profanity from being included in captions for many videos, as one might expect this blocks any audible profanity from being effectively muted.    

#### :microscope: Allow Easy Integration with OpenAI's Whisper Project and/or Custom Transcripts 
It would be great to be able to add a feature that would run a youtube url through a whisper implementation to generate accurate transcripts to be used as the source for censoring audio. 

#### :triangular_ruler: Telling my Kid No
This did not go over very well as they called into question my chosen profession and accused me of laziness  

#### :pencil: There are a ton of example projects based on Whisper popping up that allow a youtube URL as input for transcription for ex) https://app.subtitlewhisper.com/ 
Full list of related show and tell projects: https://github.com/openai/whisper/discussions/categories/show-and-tell?discussions_q=is%3Aopen+category%3A%22Show+and+tell%22+youtube&page=1

<img width="1675" alt="Screen Shot 2023-04-28 at 5 58 59 PM" src="https://user-images.githubusercontent.com/36251700/235261191-fd06a34c-bb9e-4ce5-9df6-225604884834.png">

**Comment by richardfrost** [comment]
@jbfletch Thanks for the suggestion! I haven't had a chance to look into this yet, but I'll take a look as soon as I get some time.

**Comment by han5enar** [comment]
Any updates on this? This would be such a game changer if this could work...

**Comment by richardfrost** [comment]
@han5enar Not really, unfortunately.  I thought I wrote about my finding earlier, but basically I don't think there is a good path forward from the extension, at least not right now with the current offerings. To accomplish this I think it would have to be a paid service (subscription) and would likely require quite of bit of setup, and need commitment from enough people to justify the time and cost.

I haven't found any free service that offers an API accessible to a browser extension for handling this functionality. SubtitleWhisper is interesting, and their site works pretty well, but they don't seem to provide a public API, which would be needed for the extension to be able to use it.

In summary, I don't think there can be an automatic integration at this time. That said, I do think that we can add a feature to import external captions/subtitles from a file for a video into the extension. You could use a 3rd-party service to create them if you'd like, or transcribe videos yourself, and then you could import it to the extension.

**Comment by han5enar** [comment]
Gotcha. Importing files for videos would still be astronomically better than using those awful auto subtitles lol

Thanks for updating the situation so soon, you guys do great work
